telegram/_telegramobject.py

```diff
@@ -327,6 +341,34 @@ def to_dict(self, recursive: bool = True) -> JSONDict:
             :obj:`dict`
         """
         out = self._get_attrs(recursive=recursive)
+
+        # Now we should convert TGObjects to dicts inside objects such as sequences, and convert
+        # datetimes to timestamps. This mostly eliminates the need for subclasses to override
+        # `to_dict`
+        for key, value in out.items():
+            if isinstance(value, Sequence) and not isinstance(value, str):
+                val = []  # empty list to append our converted values to
+                for item in value:
+                    if hasattr(item, "to_dict"):
+                        val.append(item.to_dict(recursive=recursive))  # type: ignore[union-attr]
+                    # This branch is useful for e.g. List[List[PhotoSize|KeyboardButton]]
+                    elif isinstance(item, (tuple, list)):
```

Comment:
why not `Sequence` here?

On second thought, we do know that all the attributes of TO are only lists (or tuples after #3249), so checking for that (both here and above) might actually suffice 🤔 I might have been caught in the "ABC" spirit 😄

Reply:
> why not `Sequence` here?

it is code from my initial attempt, before I figured I could use Sequence.

but okay, will revert to using tuple and list only